### PR TITLE
fix: google tts files are now properly reproduced from cache

### PR DESCRIPTION
**Pull request by Gusinuhe**
### Description

**Fixes**: The Mickey Mouse-like pitch when Google TTS files were used from cache.

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

Play a TTS behavior using google and cached files.

### CODE REVIEW

File: Runtime/Provider/FileTextToSpeechProvider.cs [hunk 1]
```diff
@@ -126,6 +127,20 @@ protected virtual bool IsFileCached(string filePath)
...
+        {
+            int value=0;
+            
+            for (int i=0; i<4; i++)
+            {
+                value |= bytes[offset+i] << (i*8);
+            }
+            
+            return value;
+        }
```
Comment (burnedikt) [inline Runtime/Provider/FileTextToSpeechProvider.cs:142]: formatting seems off or at least not according to our style guide

File: Runtime/Provider/FileTextToSpeechProvider.cs [hunk 2]
```diff
@@ -126,6 +127,20 @@ protected virtual bool IsFileCached(string filePath)
             return FileManager.Exists(filePath);
         }
         
+        // Found at https://answers.unity.com/questions/737002/wav-byte-to-audioclip.html
+        // static int bytesToInt(byte[] bytes,int offset=0)
```
Comment (burnedikt) [inline Runtime/Provider/FileTextToSpeechProvider.cs:131]: I think the comment in line 130 is sufficient, line 131 is rather confusing imo

**Review by burnedikt — DISMISSED** [review]

**Comment by SimonTheSourcerer** [comment]
At the beginning of each files there is a strange "sound" It's also in the uncached files but not so loud. Guess it might not be introduced by this PR but it's kinda annoying and the first time I'm aware of it.

**Comment by SimonTheSourcerer** [comment]
On the other side, tested it, does work now, no mickey mouse sounds

**Comment by Gusinuhe** [comment]
I tested it, the weird noise is present from before this PR.

**Review by SimonTheSourcerer — APPROVED** [review]

**Pull request merged**